axelor-supplychain/src/main/resources/views/Mrp.xml

```diff
@@ -80,8 +81,10 @@
 	    <view type="grid" name="mrp-line-grid"/>
 	    <view type="form" name="mrp-line-form"/>
 		<view-param name="limit" value="200"/>
-	    <domain>self.mrp.id = :_mrpId</domain>
+	    <domain>self.mrp.id = :_mrpId AND self.mrpLineType.elementSelect in (:_elementSelectIds)</domain>
 		<context name="_mrpId" expr="eval: __this__.id"/>
+		<context name="_elementSelectIds" expr="eval: [5,8,9]" if="!displayProductWithoutProposal"/>
```

Comment:
Keep a ternary operator. Also, don't give static numbers of elementSelect instead use its declared constant variables.